src/Component/Dash.js

```diff
@@ -0,0 +1,309 @@
+    justifyContent: 'flex-end',
+    padding: '0 8px',
+    ...theme.mixins.toolbar,
+  },
+  appBar: {
+    zIndex: theme.zIndex.drawer + 1,
+    transition: theme.transitions.create(['width', 'margin'], {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+  },
+  appBarShift: {
+    marginLeft: drawerWidth,
+    width: `calc(100% - ${drawerWidth}px)`,
+    transition: theme.transitions.create(['width', 'margin'], {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+  },
+  menuButton: {
+    marginRight: 36,
+  },
+  menuButtonHidden: {
+    display: 'none',
+  },
+  title: {
+    flexGrow: 1,
+  },
+  drawerPaper: {
+    position: 'relative',
+    whiteSpace: 'nowrap',
+    width: drawerWidth,
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+  },
+  drawerPaperClose: {
+    overflowX: 'hidden',
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+    width: theme.spacing(7),
+    [theme.breakpoints.up('sm')]: {
+      width: theme.spacing(9),
+    },
+  },
+  appBarSpacer: theme.mixins.toolbar,
+  content: {
+    flexGrow: 1,
+    height: '100vh',
+    overflow: 'auto',
+  },
+  container: {
+    paddingTop: theme.spacing(4),
+    paddingBottom: theme.spacing(4),
+  },
+  paper: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    overflow: 'auto',
+    flexDirection: 'column',
+  },
+  fixedHeight: {
+    height: 240,
+  },
+}));
+
+function Dash() {
+  function dateFormat(Date) {
+    let dateF = '';
+    const year = Date.getFullYear();
+    const month = Date.getMonth() + 1;
+    const day = Date.getUTCDate();
+    dateF = `${year}-${month}-${day}`;
+    return dateF;
+  }
+
+  const classes = useStyles();
+  const navigate = useNavigate();
+  const logout = () => {
+    localStorage.clear();
+    navigate('/');
+  };
+
+  const [openDrawer, setOpenDrawer] = useState(false);
+  const handleDrawerOpen = () => {
+    setOpenDrawer(true);
+  };
+  const handleDrawerClose = () => {
+    setOpenDrawer(false);
+  };
+
+  const [isAddLeaves, setAddLeaves] = useState(false);
+  const [pLeave, setPLeave] = useState(false);
+  const [date, setDate] = useState('');
+
+  const openDiag = () => {
+    setAddLeaves(true);
+  };
+
+  const closeDiag = () => {
+    setAddLeaves(false);
+  };
+
+  const handleSubmit = () => {
+    setPLeave(true);
+  };
+
+  // declare a new state variable for modal open
+  const [open, setOpen] = useState(false);
+
+  // function to handle modal open
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  // function to handle modal close
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div className={classes.root}>
```

Comment:
This component is too big, should we divide this in multiple smaller components?